frontend/dockerfile/dockerfile2llb/directives.go

```diff
@@ -53,3 +91,36 @@ func ParseDirectives(r io.Reader) map[string]Directive {
 	}
 	return m
 }
+
+func commentDirectiveParser(comment string) directiveParser {
+	return regexpDirectiveParser{
+		re: regexp.MustCompile(fmt.Sprintf(`^%s\s*([a-zA-Z][a-zA-Z0-9]*)\s*=\s*(.+?)\s*$`, comment)),
```

Comment:
I know regular expressions can be quite resource-hungry (memory, allocations); that a concern for this code?

Possible options / alternatives;

- pre-compile the regexes in an init() (?)
- wondering how "strict" we must be; for some / most of these, a `strings.HasPrefix("//") / strings.HasPrefix("#")` could satisfy most cases, and may be more lightweight; remaining code would be splitting on `=` and check if the first element is `syntax`.

Reply:
I *think* these regexes will be compiled in an `init` in a roundabout way - the only place they're called is in the global variable initialization above.

But agreed, I think we could *probably* be less strict here if we wanted - I think the directives parsing here is only really used for detecting syntax atm, the actual directive parsing will be performed by each frontend as it sees fit.

Reply:
We should actually avoid `regexp.MustCompile` in init. Better to have a `sync.Once` function that is called before these regexp are actually needed that does the initialization.

Reply:
This should be handled now :+1: